src/init.cpp

```diff
@@ -867,12 +867,10 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         nLocalServices = ServiceFlags(nLocalServices | NODE_COMPACT_FILTERS);
     }
 
-    // if using block pruning, then disallow txindex and coinstatsindex
+    // if using block pruning, then disallow txindex
     if (args.GetIntArg("-prune", 0)) {
         if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX))
             return InitError(_("Prune mode is incompatible with -txindex."));
-        if (args.GetBoolArg("-coinstatsindex", DEFAULT_COINSTATSINDEX))
-            return InitError(_("Prune mode is incompatible with -coinstatsindex."));
     }
```

Comment:
IMO this should be a separate PR so we don't have to review the whole coinstatsindex logic for prune-safety in this PR.

Reply:
Hmm, this part was, to me, the main motivation for this PR. Given the review that has already gone into it up until now with this included I think it's ok to leave it in. But will reconsider if others agree with you.